### ISSUE TITLE
[Question] Type URI as string in model

### ISSUE THREAD

**Issue by sverhagen** [springfox/springfox#2059]
Please take the time to search the repository, if your question has already been asked or answered.

- [ ] What version of the library are you using? **2.7.0** Is it the latest version? **yes**

What kind of issue is this?

 - [x] Question. Is this a question about how to do a certain thing? **yes, for now**

 - [ ] Bug report. **no, probably not**
 - [ ] Feature Request. **no**

Here's the question, I've created the following request object:

    public class EnvironmentUpdateRequest {
        @NotNull
        @ApiModelProperty(dataType = "string") // an attempt to fix my issue
        private URI consulUri; // that's java.net.URI
    
        public URI getConsulUri() {
            return consulUri;
        }
    
        public void setConsulUri(URI consulUri) {
            this.consulUri = consulUri;
        }
    }

It appears as follows in the emitted Swagger JSON:

    "EnvironmentUpdateRequest": {
        "properties": {
            "consulUri": {
                "$ref": "#/definitions/URI"
            }
        },
        "type": "object"
    },

Since it's `"type": "object"`, and there's a comprehensive description of `URI` in `definitions`, it'll use that. But in reality, Jackson perfectly serializes a string into the desired `java.net.URI` that's needed. So, I'd say that my API works fine, but its documentation is flawed. How can I make Springfox type this field as `string`?

Thank you for an awesome library!

**Comment by dilipkrish** [comment]
I think as a workaround you could create a rule to map `URI` to `String` but I agree that it should just work out of the box.

**Comment by sverhagen** [comment]
This seems to work, and I'll move forward with this as a workaround:

    .alternateTypeRules(
        newRule(typeResolver.resolve(URI.class), typeResolver.resolve(String.class)))

In combination with:

    @Autowired
    private TypeResolver typeResolver;

I don't think it's necessarily a bug that `URI` isn't mapped as `string`, but it is strange that `dataType = "string"` isn't adhered to. (No matter Springfox "Philosophically" wanting "to discourage using (swagger-core) annotations".)

Thanks for the quick response.

**Comment by dilipkrish** [comment]
Actually it will work if you specified `java.lang.String`. dataType is [ignored when not fully qualified](http://springfox.github.io/springfox/docs/current/#overriding-property-datatypes).

**Comment by sverhagen** [comment]
Please note that I'm spelling `string`, not `String`. I was under the impression that `string` would be considered the _primitive_, and thus valid in the form as given. I'll quote the following links, although I'm unsure whether either of those is all too relevant to Springfox: ["This can be the class name or a primitive"](http://docs.swagger.io/swagger-core/current/apidocs/index.html?io/swagger/annotations/ApiModelProperty.html), and [types from JSON Schema Specification](https://swagger.io/specification/#data-types-13). I _do_ agree that `dataType = "java.lang.String"` works, I hadn't tried that, my bad entirely! Thanks again.

**Comment by dilipkrish** [comment]
@sverhagen Yeah there are slight deviations when using the annotations from the mother ship :) Mostly to protect library users from fat-fingering errors.
